src/main/java/org/sagebionetworks/bridge/models/schedules2/adherence/AdherenceState.java

```diff
@@ -0,0 +1,188 @@
+        streamsByStreamKey = new HashMap<>();
+        adherenceByGuid = builder.adherenceRecords.stream()
+                .collect(toMap(AdherenceRecord::getInstanceGuid, (a) -> a));
+        
+        LocalDate localNow = null;
+        zone = null;
+        if (builder.now != null) {
+            localNow = builder.now.toLocalDate();
+            zone = builder.now.getZone();    
+        }
+        // however, use the subject’s preferred time zone if available
+        if (clientTimeZone != null) {
+            zone = DateTimeZone.forID(clientTimeZone);
+        }
+        
+        // LocalDate localNow = (builder.now != null) ? builder.now.toLocalDate() : null;
+        for (StudyActivityEvent event : builder.events) {
+            // DateTime eventTimestamp = event.getTimestamp();
+            DateTime eventTimestamp = event.getTimestamp().withZone(zone);
+            int daysSince = Days.daysBetween(eventTimestamp.toLocalDate(), localNow).getDays();
+            daysSinceEventByEventId.put(event.getEventId(), daysSince);
+            eventTimestampByEventId.put(event.getEventId(), eventTimestamp);
+        }
+    }
+    
+    public List<TimelineMetadata> getMetadata() {
+        return metadata;
+    }
+    public boolean showActive() {
+        return showActive;
+    }
+    public DateTime getNow() {
+        return now;
+    }
+    public String getClientTimeZone() {
+        return clientTimeZone;
+    }
+    public EventStream getEventStreamById(String eventId) {
+        EventStream stream = streamsByEventId.get(eventId);
+        if (stream == null) {
+            stream = new EventStream();
+            stream.setStartEventId(eventId);
+            stream.setEventTimestamp(eventTimestampByEventId.get(eventId));
+            streamsByEventId.put(eventId, stream);
+        }
+        return stream;
+    }
+    public EventStreamDay getEventStreamDayByKey(TimelineMetadata meta) {
+        String streamKey = String.format("%s:%s:%s", meta.getSessionGuid(), 
+                meta.getSessionStartEventId(), meta.getSessionInstanceStartDay());
+        EventStreamDay eventStreamDay = streamsByStreamKey.get(streamKey);
+        
+        if (eventStreamDay == null) {
+            int startDay = meta.getSessionInstanceStartDay();
+            String eventId = meta.getSessionStartEventId();
+            
+            eventStreamDay = new EventStreamDay();
+            eventStreamDay.setSessionGuid(meta.getSessionGuid());
+            eventStreamDay.setSessionName(meta.getSessionName());
+            eventStreamDay.setSessionSymbol(meta.getSessionSymbol());
+            eventStreamDay.setWeek(startDay / 7);
+            eventStreamDay.setStudyBurstId(meta.getStudyBurstId());
+            eventStreamDay.setStudyBurstNum(meta.getStudyBurstNum());
+            streamsByStreamKey.put(streamKey, eventStreamDay);
+            getEventStreamById(eventId).addEntry(startDay, eventStreamDay);
+        }
+        return eventStreamDay;
+    }
```

Comment:
Mixes two things: an immutable parameter object, and an object for collecting the calculated state of streams. This is confusing and the stream calculation should go back into the generator (which would not be an instance in that case, as before).

Reply:
After discussing I decided to leave this. From the perspective of the generators, these do not appear mutable, it just appears that the state has a stream, for every ID. We could create them all up front, but there's just no reason.